src/test/java/com/google/firebase/remoteconfig/FirebaseRemoteConfigClientImplTest.java

```diff
@@ -0,0 +1,377 @@
+          MockLowLevelHttpResponse mockResponse, HttpResponseInterceptor interceptor) {
+    MockHttpTransport transport = new MockHttpTransport.Builder()
+            .setLowLevelHttpResponse(mockResponse)
+            .build();
+
+    return FirebaseRemoteConfigClientImpl.builder()
+            .setProjectId("test-project")
+            .setJsonFactory(Utils.getDefaultJsonFactory())
+            .setRequestFactory(transport.createRequestFactory())
+            .setChildRequestFactory(Utils.getDefaultTransport().createRequestFactory())
+            .setResponseInterceptor(interceptor)
+            .build();
+  }
+
+  private FirebaseRemoteConfigClientImpl initClientWithFaultyTransport() {
+    HttpTransport transport = TestUtils.createFaultyHttpTransport();
+    return FirebaseRemoteConfigClientImpl.builder()
+            .setProjectId("test-project")
+            .setJsonFactory(Utils.getDefaultJsonFactory())
+            .setRequestFactory(transport.createRequestFactory())
+            .setChildRequestFactory(Utils.getDefaultTransport().createRequestFactory())
+            .build();
+  }
+
+  private FirebaseRemoteConfigClientImpl.Builder fullyPopulatedBuilder() {
+    return FirebaseRemoteConfigClientImpl.builder()
+            .setProjectId("test-project")
+            .setJsonFactory(Utils.getDefaultJsonFactory())
+            .setRequestFactory(Utils.getDefaultTransport().createRequestFactory())
+            .setChildRequestFactory(Utils.getDefaultTransport().createRequestFactory());
+  }
+
+  private void checkGetRequestHeader(HttpRequest request) {
+    assertEquals("GET", request.getRequestMethod());
+    assertEquals(TEST_RC_URL, request.getUrl().toString());
+    HttpHeaders headers = request.getHeaders();
+    assertEquals("2", headers.get("X-GOOG-API-FORMAT-VERSION"));
+    assertEquals("fire-admin-java/" + SdkUtils.getVersion(), headers.get("X-Firebase-Client"));
+    assertEquals("gzip", headers.getFirstHeaderStringValue("Accept-Encoding"));
+  }
+
+  private void checkExceptionFromHttpResponse(
+          FirebaseRemoteConfigException error,
+          ErrorCode expectedCode) {
+    checkExceptionFromHttpResponse(error, expectedCode, null, "test error");
+  }
+
+  private void checkExceptionFromHttpResponse(
+          FirebaseRemoteConfigException error,
+          ErrorCode expectedCode,
+          RemoteConfigErrorCode expectedRemoteConfigCode,
+          String expectedMessage) {
+    assertEquals(expectedCode, error.getErrorCode());
+    assertEquals(expectedMessage, error.getMessage());
+    assertTrue(error.getCause() instanceof HttpResponseException);
+    assertEquals(expectedRemoteConfigCode, error.getRemoteConfigErrorCode());
+
+    assertNotNull(error.getHttpResponse());
+    OutgoingHttpRequest request = error.getHttpResponse().getRequest();
+    assertEquals(HttpMethods.GET, request.getMethod());
+    assertTrue(request.getUrl().startsWith("https://firebaseremoteconfig.googleapis.com"));
```

Comment:
`.equals(TEST_RC_URL)`?

Reply:
The goal is to reuse this function for other operations :)
We will have different complete URLS for them. I need to figure out the best way to group repeating tests...